ChainSafe/web3.unity
# prefab creation

I've created prefabs to ease the learning curve for new developers, this also leaves the samples untouched so they can be used as a point of reference as needed depending on the developers coding experience. Showing users a simple prefab with the block of code they're after is going to save us a lot of time in the long run when it comes to teaching the SDK & customer support.

Below I've listed the issues this fixes as well as a how to guide for testing (Thanks Juan!)

1. Problem: The example scene works great in terms of showing a working call. However, we can't alter this code.

1. Fix: Created prefab & script folders. Each each function now has a corresponding prefab and script in `web3.unity\src\UnitySampleProject\Assets\Samples\web3.unity SDK\2.5.0` You may copy the code to a new script and make changes without having to worrying about breaking anything.
![image](https://github.com/ChainSafe/web3.unity/assets/57473220/cd275459-9ad0-48e0-8404-03d16a25eee1)

2. Problem: Editor inputs are clashing with my script inputs.

2. Fix: I've made the inputs on the prefabs private so the user is only editing code in 1 place (the script). This removes the possibility of human error and saves us some time in discord by ruling it out.
![image](https://github.com/ChainSafe/web3.unity/assets/57473220/6f7348c8-402e-4e25-b6ac-80703cd663cb)

3. Problem: The example scripts are hard to find as they are deep in the menu hierarchy and they only work on a button.

3. Fix: Easy fix with the new examples! As I've added monobehaviour and removed the sample behavior, i was able to move the script onto the root of the object. This makes it easier for new users to find and make changes to the script.
![image](https://github.com/ChainSafe/web3.unity/assets/57473220/9c70c6cf-04d1-4df0-8c68-06a858db144e)

4. Problem: This SDK is hard to understand

4. Fix: Not anymore! It's now a 3 step process! This will also be pushed to the docs.
- Find and copy the prefab script in the very easy to find prefab folder
- Alter the class name and input variables
- Do something with the results. Simple!

To test these prefabs:

1. Place the sample login scene as your first scene and then a 2nd blank scene in your build settings.
2. Make sure your 2nd scene has an event system present (if we put the event system in the prefabs it loops when > 1)
3. Place as many prefabs as you like in the 2nd scene and test webgl/desktop (they stack on each other and have a close button)
4. The prefabs should resize dynamically for any screen size (mobile too) and results will be logged to the console.
![image](https://github.com/ChainSafe/web3.unity/assets/57473220/ea45b259-4ad2-4815-895f-66ec77cb121b)

I had to comment out the walletconnectmodal in login.cs as it wasn't happy during my testing. Leaving a note here just incase.
